core/src/test/java/com/linecorp/armeria/client/retry/RetryingClientTest.java

```diff
@@ -440,20 +440,26 @@ void retryOnResponseTimeout() {
 
     @Test
     void retryWithContentOnResponseTimeout() {
+        final AtomicInteger counter = new AtomicInteger();
         final Backoff backoff = Backoff.fixed(100);
         final RetryRuleWithContent<HttpResponse> strategy =
                 RetryRuleWithContent.<HttpResponse>onResponse((unused, response) -> {
+                    counter.incrementAndGet();
                     return response.aggregate().thenApply(unused0 -> false);
                 }).orElse(RetryRuleWithContent.onResponse((unused, response) -> {
+                    counter.incrementAndGet();
                     return response.aggregate().thenApply(unused0 -> false);
                 })).orElse(RetryRuleWithContent.<HttpResponse>onResponse((unused, response) -> {
+                    counter.incrementAndGet();
                     return response.aggregate().thenApply(unused0 -> false);
                 }).orElse(RetryRule.builder()
                                    .onException(ResponseTimeoutException.class)
                                    .thenBackoff(backoff)));
         final WebClient client = client(strategy, 0, 500, 100);
         final AggregatedHttpResponse res = client.get("/1sleep-then-success").aggregate().join();
         assertThat(res.contentUtf8()).isEqualTo("Succeeded after retry");
+        // Make sure that all customized RetryRuleWithContents are called.
+        assertThat(counter.get()).isEqualTo(3);
```

Comment:
This may mean the first rule was evaluated 3 times. How about using a `BlockingQueue`?

Reply:
That's a good suggestion. 👍